pkg/controller/registry/resolver/step_resolver.go

```diff
@@ -193,14 +193,32 @@ func (r *OperatorStepResolver) ResolveSteps(namespace string) ([]*v1alpha1.Step,
 			})
 		}
 
-		// add steps for subscriptions for bundles that were added through resolution
+		// add steps for subscriptions for bundles that were added through resolution and bundles that have been marked deprecated
 		for sub := range existingSubscriptions {
-			if sub.Status.CurrentCSV == op.Name {
-				continue
+			updated := false
+			if len(op.Deprecations) == 0 && corev1.ConditionStatus(sub.Status.GetCondition(v1alpha1.SubscriptionOperatorDeprecated).Type) != corev1.ConditionUnknown {
```

Comment:
This seems like it deserves a comment since it is non-obvious that `GetCondition` will return a populated condition set to `Unknown` when the type is not found.

However, do we even need the `Unknown` check? Can we just always remove it if there are no deprecations?

Reply:
Yeah this func is a bit odd in how it behaves. We could also use a more generic func for this op but I believe we will have to do conversions to a generic condition type.

The reason I did the check was primarily to avoid doing a no-op removal then adding the subscription to the list of `updatedSubs` when potentially nothing has changed. If we don't care about that though we can just do it regardless.

I'll also note however I did check `Type` here instead of `Status` so I will fix that :+1: